Sources/swift-openapi-generator/GenerateCommand.swift

```diff
@@ -42,16 +42,19 @@ struct _GenerateCommand: AsyncParsableCommand {
     )
     var outputDirectory: URL = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
 
-    @Flag(
+    @Option(
         help:
-            "Whether this invocation is from the SwiftPM plugin. We always need to produce all files when invoked from the plugin. Non-requested modes produce empty files."
+            "Source of invocation if by a plugin. The generator needs to produce all files when invoked as a build plugin, so non-requested modes produce empty files."
     )
-    var isPluginInvocation: Bool = false
+    var invokedFrom: PluginSource?
```

Comment:
Just bumping the first half of this comment: https://github.com/apple/swift-openapi-generator/pull/98#discussion_r1265064137

Let's also rename the option (will require a change in the plugins as well, and in the docs) to `--plugin-source`, I'm not a fan of options ending with a preposition.

```suggestion
    var pluginSource: PluginSource?
```